src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -415,6 +431,50 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
             })
             .collect())
     }
+
+    fn watch_services(&self) -> BoxStream<'static, Result<ServiceEvent, anyhow::Error>> {
+        fn into_service_event(pod: Pod) -> Result<ServiceEvent, anyhow::Error> {
+            let pod_name = pod.metadata.name.unwrap();
+            let pod_labels = pod.metadata.labels.unwrap();
+            let pod_status = pod.status.unwrap();
```

Comment:
Is there guaranteed to be a status? This isn't in the `Resource` or `ResourceExt` traits, which are typically the stuff that is guaranteed to exist when querying resources.

Side note, if you want to avoid some `unwrap()`s, those traits are pretty nice. (I think they just do the unwrap for you, though).

Reply:
I couldn't find out which fields are guaranteed to be set and which are not, so I just went with the reasoning that surely a pod would always have a status. Thanks for catching this!